libs/payments/cart/src/lib/checkout.service.ts

```diff
@@ -389,25 +389,31 @@ export class CheckoutService {
             off_session: false,
           }
         );
-        stripeIntentId = intent.id;
       } else {
         intent = await this.setupIntentManager.createAndConfirm(
           customer.id,
           confirmationTokenId
         );
-        stripeIntentId = intent.id;
       }
     } catch (error) {
-      // At least update the cart with the subscription ID before throwing
-      await this.cartManager.updateFreshCart(cart.id, version, {
-        stripeSubscriptionId: subscription.id,
-      });
-      throw error;
+      if (error?.payment_intent) {
+        intent = error.payment_intent;
+      } else if (error?.setup_intent) {
+        intent = error.setup_intent;
+      }
+
+      if (!intent) {
+        // At least update the cart with the subscription ID before throwing
+        await this.cartManager.updateFreshCart(cart.id, version, {
+          stripeSubscriptionId: subscription.id,
+        });
+        throw error;
+      }
```

Comment:
No error will be thrown if `intent` is truthy. Is this intended?

Reply:
Yup. If the error thrown contained the intent object, it's handled in the intent handling logic below.